extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -137,6 +143,25 @@ void validateConfigProperties(ConfigRecorder recorder, List<ConfigPropertyBuildI
         recorder.validateConfigProperties(propNamesToClasses);
     }
 
+    @BuildStep
+    BeanRegistrarBuildItem registerConfigRootsAsBeans(ConfigurationBuildItem configItem) {
+        return new BeanRegistrarBuildItem(new BeanRegistrar() {
+            @Override
+            public void register(RegistrationContext context) {
+                for (RootDefinition rootDefinition : configItem.getReadResult().getAllRoots()) {
+                    if (rootDefinition.getConfigPhase() == ConfigPhase.BUILD_AND_RUN_TIME_FIXED
+                            || rootDefinition.getConfigPhase() == ConfigPhase.RUN_TIME) {
+                        context.configure(rootDefinition.getConfigurationClass()).types(rootDefinition.getConfigurationClass())
+                                .scope(Dependent.class).creator(mc -> {
```

Comment:
It might be a stupid question but why a `Dependent` scope?

Reply:
Well, we could use `@Singleton` but since we don't create any instance but just delegate to a static field it does not make sense to store the reference in the context. Having `@Dependent` means that `Contextual.create()` is invoked for every injetion point. In our case, `Config.ApplicationConfig` is read instead of context lookup (i.e. `Map.get()` for singleton).

And normal scopes don't make sense either. A client proxy would be useless (one more useless delegation layer).